pandas/core/series.py

```diff
@@ -580,8 +581,19 @@ def __arrow_c_stream__(self, requested_schema=None):
         PyCapsule
         """
         pa = import_optional_dependency("pyarrow", min_version="16.0.0")
-        ca = pa.chunked_array([pa.Array.from_pandas(self, type=requested_schema)])
-        return ca.__arrow_c_stream__(requested_schema)
+        type = (
+            pa.DataType._import_from_c_capsule(requested_schema)
+            if requested_schema is not None
+            else None
+        )
+        if isinstance(self.dtype, ArrowDtype):
+            # fastpath!
+            ca = self.values._pa_array
+            if type is not None:
+                ca = ca.cast(type)
+        else:
+            ca = pa.chunked_array([pa.Array.from_pandas(self, type=type)])
```

Comment:
This does not yet handle StringDtype (which is not an ArrowDtype). 

Alternative option is something like (untested):

```
ca = pa.array(self, type=type)
if not isinstance(ca, pa.ChunkedArray):
    ca = pa.chunked_array([ca])
```

Note: 1) it's a bit strange, but `pa.array()` can return a chunked array (in this case, it will check the `__arrow_array__` attribute on the passed object, and for our dtypes backed by a chunked array, that method will return the chunked array, and this is directly passed through as result of `pa.array()`), and 2) `pa.array()` is essentially just the same as `pa.Array.from_pandas` but a bit simpler (and because we rely on it returning potentially a chunked array, it would read even more strangely ..)

Reply:
that seems to work, thanks @jorisvandenbossche !